utils/ahp.py

```diff
@@ -44,15 +33,36 @@ def randomize_matrix(n: int) -> np.ndarray:
     return matrix
 
 
-# przykład jak to działa dla losowych odpowiedzi
-if __name__ == "__main__":
-    test = AnalyticalHierarchyProcess()
-    test.add_criteria_matrix(randomize_matrix(3))
-    test.add_pets_matrix("time", randomize_matrix(3))
-    test.add_pets_matrix("cost", randomize_matrix(3))
-    test.add_pets_matrix("lifespan", randomize_matrix(3))
+class AnalyticalHierarchyProcess:
+    def __init__(self, pets_data: List[Dict] = None, criteria_data: List[Dict] = None):
+        if criteria_data is None:
+            self.criteria = ["time", "cost", "lifespan"]
+        else:
+            self.criteria = [criteria['name'] for criteria in criteria_data]
+
+        if pets_data is None:
+            self.pets = ["cat", "dog", "snake"]
+        else:
+            self.pets = [pet['name'] for pet in pets_data]
+
+        self.criteria_matrix = np.ones([len(self.criteria), len(self.criteria)])
+        self.pets_matrices = [np.ones([len(self.pets), len(self.pets)]) for _ in self.criteria]
```

Comment:
```suggestion
class AnalyticalHierarchyProcess:
    def __init__(self, pets_data: list[dict[str, str]], criteria_data: list[dict[str, str]]):
        self.criteria = [criteria['name'] for criteria in criteria_data]
        self.pets = [pet['name'] for pet in pets_data]

        self.criteria_matrix = np.ones([len(self.criteria), len(self.criteria)])
        self.pets_matrices = [np.ones([len(self.pets), len(self.pets)]) for _ in self.criteria]
```
I don't think that there's need to use some random default categories. Also, referring to the comment about method that checks if the matrices have been set, it's probably better to not initialise it to a meaningful value, but that's discussable.

Reply:
hmm, yeah that would make sense